File: Problems/1201.-1300./1299._Replace_Elements_with_Greatest_Element_on_Right_Side/ReplaceElementsWithGreatestElementOnRightSide0.py
class Solution:
  def replaceElements(self, arr: List[int]) -> List[int]:
    n = len(arr)
    if (n == 1): return [-1]
        
    # Replace the last element with -1
    temp = arr[n-1]; arr[n-1] = -1
    
    # Replace the 2nd last element with the last one
    max_value = max(temp, arr[n-2])
    arr[n-2] = temp
    
    # Continue with loop
    for i in range(n-3,-1,-1):
        temp = arr[i]
        arr[i] = max_value
        max_value = max(temp, max_value)
    return arr

File: Problems/1201.-1300./1299._Replace_Elements_with_Greatest_Element_on_Right_Side/ReplaceElementsWithGreatestElementOnRightSide0.java
//!/bin/java

class Solution {
  public int[] replaceElements(int[] arr) {
    if (arr.length == 1) return new int[]{-1};
        
    // Replace the last element with -1
    int temp = arr[arr.length-1];
    arr[arr.length-1] = -1;
    
    // Replace the 2nd last elements with last element
    int maxVal = Math.max(temp,arr[arr.length-2]);
    arr[arr.length-2] = temp;
    
    // Continue with loop
    for (int i = arr.length-3; i >= 0; i--) {
        temp = arr[i];
        arr[i] = maxVal;
        maxVal = Math.max(temp,maxVal);
    }
    return arr;
  }
}
